Marlin/ultralcd.cpp

```diff
@@ -1285,6 +1279,18 @@ void kill_screen(const char* lcd_msg) {
       }
     }
 
+    inline void lcd_wait_for_homing() {
+      #if ENABLED(MANUAL_BED_LEVELING)
+        if (axis_homed[X_AXIS] && axis_homed[Y_AXIS] && axis_homed[Z_AXIS])
+          lcd_goto_screen(_lcd_level_bed_homing_done);
+      #else
+        if (no_reentrance) return;
+        while (!axis_homed[X_AXIS] || !axis_homed[Y_AXIS] || !axis_homed[Z_AXIS]) idle();
+        no_reentrance = true;
```

Comment:
I also think that wait_for_homing should not have side effects like changing the screen